pkg/controller/kubefedwebhook/openshift/openshift.go

```diff
@@ -0,0 +1,276 @@
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+)
+
+const (
+	caBundleConfigMapName = "config-service-ca"
+	serviceSecretKey      = "service-ca.crt"
+	serviceAccountName    = "system:serviceaccount:kubefed-admission-webhook"
+	sccName               = "privileged"
+)
+
+var (
+	extension = common.Extension{
+		Transformers:  []mf.Transformer{deploymentController, annotateWebhookService},
+		PreInstalls:   []common.Extender{caBundleConfigMap, addUserToSCC},
+		PostInstalls:  []common.Extender{ensureCaBundleIsPresent},
+		Transclosures: []common.Transclosure{addCaBundleToWebhookConfiguration},
+	}
+	log    = logf.Log.WithName("openshift")
+	api    client.Client
+	scheme *runtime.Scheme
+)
+
+func IsRunningOpenshift(c client.Client) (bool, error) {
+	if routeExists, err := anyKindExists(c, "", schema.GroupVersionKind{"route.openshift.io", "v1", "route"}); err != nil {
+		return false, err
+	} else if !routeExists {
+		// Not running in OpenShift
+		return false, nil
+	} else {
+		return true, nil
+	}
+}
+
+// Configure OpenShift if we're soaking in it
+func Configure(c client.Client, s *runtime.Scheme) (*common.Extension, error) {
+	if routeExists, err := anyKindExists(c, "", schema.GroupVersionKind{"route.openshift.io", "v1", "route"}); err != nil {
+		return nil, err
+	} else if !routeExists {
+		// Not running in OpenShift
+		return nil, nil
+	}
+
+	// Register scheme
+	if err := configv1.Install(s); err != nil {
+		log.Error(err, "Unable to register scheme")
+		return nil, err
+	}
+
+	api = c
+	scheme = s
+	return &extension, nil
+}
+
+func addUserToSCC(instance *kubefedv1alpha1.KubeFedWebHook) error {
+	scc := &unstructured.Unstructured{}
+	scc.SetAPIVersion("security.openshift.io/v1")
+	scc.SetKind("SecurityContextConstraints")
+
+	err := api.Get(context.TODO(), client.ObjectKey{Name: sccName}, scc)
+	if err != nil {
+		return err
+	}
+	// Verify if SA has already been assigned to the SCC
+	existing, exists, _ := unstructured.NestedStringSlice(scc.UnstructuredContent(), "users")
+	if exists {
+		for _, e := range existing {
+			if e == serviceAccountName {
+				return nil
+			}
+		}
+		existing = append(existing, serviceAccountName)
+	}
+
+	unstructured.SetNestedStringSlice(scc.UnstructuredContent(), existing, "users")
+	err = api.Update(context.TODO(), scc)
+	if err != nil {
+		return err
+	}
+	log.Info(fmt.Sprintf("Added ServiceAccount %q to SecurityContextConstraints %q", serviceAccountName, sccName))
+	return nil
+}
+
+func annotateWebhookService(u *unstructured.Unstructured) error {
+	if u.GetKind() == "Service" && u.GetName() == "kubefed-admission-webhook" {
+		service := &v1.Service{}
+		if err := scheme.Convert(u, service, nil); err != nil {
+			return err
+		}
+		service.Annotations["service.beta.openshift.io/serving-cert-secret-name"] = "kubefed-admission-webhook-serving-cert"
+	}
+	return nil
+}
+
+func deploymentController(u *unstructured.Unstructured) error {
```

Comment:
Per our in-person conversation, we do not need to expose the CA bundle to the webhook controller.